xarray/testing/assertions.py

```diff
@@ -418,25 +433,20 @@ def _assert_variable_invariants(var: Variable, name: Hashable = None):
 
 
 def _assert_dataarray_invariants(da: DataArray, check_default_indexes: bool):
-    assert isinstance(da._variable, Variable), da._variable
-    _assert_variable_invariants(da._variable)
+    _assert_variable_invariants(da._variable, name=da.name, check_default_indexes=False)
 
     assert isinstance(da._coords, dict), da._coords
-    assert all(isinstance(v, Variable) for v in da._coords.values()), da._coords
 
     if check_default_indexes:
         assert all(set(v.dims) <= set(da.dims) for v in da._coords.values()), (
             da.dims,
             {k: v.dims for k, v in da._coords.items()},
         )
-        assert all(
-            isinstance(v, IndexVariable)
-            for (k, v) in da._coords.items()
-            if v.dims == (k,)
-        ), {k: type(v) for k, v in da._coords.items()}
 
     for k, v in da._coords.items():
-        _assert_variable_invariants(v, k)
+        _assert_variable_invariants(
+            v, k, check_default_indexes=check_default_indexes, is_index=k in da._indexes
```

Comment:
```suggestion
            v, k, is_index=k in da._indexes
```
"check default indexes" doesn't mean anything for Variable, which is only data + named dims + attrs